### PR TITLE
Docs/getting started md update

### PR DESCRIPTION
**Pull request by xsalonx**
#### I DON'T have JIRA ticket
- [x] explain what this PR does
- [x] documentation was updated or added

Notable changes for developers:
- update of docs

### CODE REVIEW

File: docs/GETTING-STARTED.md [hunk 1]
```diff
@@ -1,43 +1,34 @@
-# Running the application
 
-## Npm tasks desciption
+## Running the application
```
Comment (Ehevi) [inline docs/GETTING-STARTED.md:2]: Please use the standard formatting.

File: docs/GETTING-STARTED.md [hunk 2]
```diff
@@ -1,43 +1,34 @@
...
-
-### Database dumps managing, dumps are stored in `<ProjectDir>/database/cache/dumps`
-9. `dump:make <DUMP_NAME>` - make a data dump and store it under the specified name
-10. `dump:list` - list all the dumps stored in local cache
-11. `dump:restore <DUMP_NAME>` - clean the db and restore the data from the specified dump file
-12. `dump:remove <DUMP_NAME>` - remove the dump file from cache
-
-### Testing
+#### Testing
+15. `test` - main task for testing purpose, run static analysis and codecov on docker containers, results are available in `<ProjectDir>/reports/`
```
Comment (Ehevi) [inline docs/GETTING-STARTED.md:19]: If you want to have a numbered list here, please at least make sure that the numbers are in order.

File: docs/GETTING-STARTED.md [hunk 3]
```diff
@@ -1,43 +1,35 @@
+
 # Running the application
 
-## Npm tasks desciption
+Most tasks are wrappers for `rctmake`.
+
+#### Npm tasks desciption
```
Comment (Ehevi) [inline docs/GETTING-STARTED.md:5]: Please stick to the standards (`##`, not `####`)

File: docs/GETTING-STARTED.md [hunk 4]
```diff
@@ -1,43 +1,35 @@
...
 Npm tasks are run via 
 ```bash
-npm run <TASK>
+npm run <TASK> -- RCTMAKE_ADDITIONAL_ARGUMENTS
 ```
+#### Within docker:
```
Comment (Ehevi) [inline docs/GETTING-STARTED.md:10]: `###` not `####`

Comment (Ehevi) [reply docs/GETTING-STARTED.md:10]: `:` at the end is not needed

File: docs/GETTING-STARTED.md [hunk 5]
```diff
@@ -1,43 +1,35 @@
...
-
-2. `dev` - prune, build and run containers, one for the database and one for the application. Changes in the local repo force the application restart on docker (see nodemon and the `start:dev` script).
-Dev does not deploy the new database, it only clears it. In case of any changes in the database or docker images, the use of `dev:prune` is advised.
-3. `dev:up` - up stopped (previously) built containers (no data in db is erased)
-4. `dev:up-r <DUMP_NAME>`  - as dev:up but also restore specified dump (existing in `<ProjectDir>/database/cache/dumps`), see the `dump:list` task
-5. `app:attach` - attach to the application container console
-6. `db:check` - export the current pgmodeler design as the database definition file, delete the database container and rebuild it with the new design. Application container will be deleted as well.
-7. `db:attach` - run psql as postgres (connected to RCT database) on database container
-8. `db:clean` - clean database in the running database docker
+#### Starting locally:
```
Comment (Ehevi) [inline docs/GETTING-STARTED.md:14]: `:` should be removed

File: docs/GETTING-STARTED.md [hunk 6]
```diff
@@ -1,43 +1,35 @@
...
-
-2. `dev` - prune, build and run containers, one for the database and one for the application. Changes in the local repo force the application restart on docker (see nodemon and the `start:dev` script).
-Dev does not deploy the new database, it only clears it. In case of any changes in the database or docker images, the use of `dev:prune` is advised.
-3. `dev:up` - up stopped (previously) built containers (no data in db is erased)
-4. `dev:up-r <DUMP_NAME>`  - as dev:up but also restore specified dump (existing in `<ProjectDir>/database/cache/dumps`), see the `dump:list` task
-5. `app:attach` - attach to the application container console
-6. `db:check` - export the current pgmodeler design as the database definition file, delete the database container and rebuild it with the new design. Application container will be deleted as well.
-7. `db:attach` - run psql as postgres (connected to RCT database) on database container
-8. `db:clean` - clean database in the running database docker
+## Starting locally:
```
[the remark this answers is not in the document]
Comment (xsalonx) [reply docs/GETTING-STARTED.md]: Done

File: docs/GETTING-STARTED.md [hunk 7]
```diff
@@ -1,43 +1,35 @@
...
 
+Most tasks are wrappers for `rctmake`.
+
 ## Npm tasks desciption
 Npm tasks are run via 
 ```bash
-npm run <TASK>
+npm run <TASK> -- RCTMAKE_ADDITIONAL_ARGUMENTS
 ```
+## Within docker:
```
[the remark this answers is not in the document]
Comment (xsalonx) [reply docs/GETTING-STARTED.md]: Done

File: docs/GETTING-STARTED.md (hunk 5 above)
Comment (xsalonx) [reply docs/GETTING-STARTED.md]: Done

File: docs/GETTING-STARTED.md (hunk 4 above)
Comment (xsalonx) [reply docs/GETTING-STARTED.md]: Done

File: docs/GETTING-STARTED.md (hunk 2 above)
Comment (xsalonx) [reply docs/GETTING-STARTED.md]: Done

File: docs/GETTING-STARTED.md (hunk 1 above)
Comment (xsalonx) [reply docs/GETTING-STARTED.md]: Done

File: docs/GETTING-STARTED.md [hunk 8]
```diff
@@ -1,43 +1,24 @@
 # Running the application
 
-## Npm tasks desciption
 Npm tasks are run via 
 ```bash
-npm run <TASK>
+npm run <TASK> -- RCTMAKE_ADDITIONAL_ARGUMENTS
 ```
+Most tasks are wrappers for `rctmake`.
```
Comment (Ehevi) [inline docs/GETTING-STARTED.md:7]: ➡️ Most tasks are `rctmake` wrappers. Run `rctmake --help` for details.

File: docs/GETTING-STARTED.md [hunk 9]
```diff
@@ -1,43 +1,24 @@
...
-npm run <TASK>
+npm run <TASK> -- RCTMAKE_ADDITIONAL_ARGUMENTS
 ```
+Most tasks are `rctmake` wrappers.
 
-### Within docker:
-1. `prune` - removes each container related to this project
+## Starting with docker
+1. `dev` - main task for running application in dev mode. The task prunes, builds and runs containers, one for the database and one for the application. Changes in the local repo force the application restart (see nodemon and the `start:dev` script, see also `rctmake` option `--target-modifier`). You can run multiple instances of application in dev mode simultaneously, see `rctmake` option `--subtarget` and `--rct-http-port`.
+2. other `*dev*` tasks are modification of pervious one.
```
Comment (Ehevi) [inline docs/GETTING-STARTED.md:11]: why the `*`?

File: docs/GETTING-STARTED.md [hunk 10]
```diff
@@ -1,43 +1,24 @@
...
+2. `deploy:db:local` - deploy the database locally with the use of `<ProjectDir>/database/setup-db.sh` script. It uses the same env as the upper task and follows the same env vars substitution logic.
 
 ## Using grid certificates
 Grid certificates need to be located in `<ProjectDir>/security/`. The certificate used by the application has to be named (TMP *TODO filename convention*) `rct-alimonitor-cert.p12`. It is also required to set `ALIMONITOR_PASSPHRASE` env var which holds the passphrase to that certificate.
 
 ## Reaching CERN network
-1. Default behaviour of the application running on docker (var `RUNNING_ENV` is set to `DOCKER`) is to use ssh proxy `socks://172.200.200.1:12345` opened on a host, so you need to locally open ssh socket via command:
+1. By default, the application running on docker (var `RUNNING_ENV` is set to `DOCKER`) doesn't use proxy. Some functions of app (sync data) requires acces to cern network, so if running outside its, you can use some proxy configured by env var `CERN_SOCKS`, by defualt `socks://172.200.200.1:12345`, for wich you should open ssh socket locally via command:
```
Comment (Ehevi) [inline docs/GETTING-STARTED.md:21]: ## Reaching CERN network
Crucial functionalities (e.g. data synchronization) require access to the CERN network. If the host is not directly connected, a proxy configuration is needed.

Comment (xsalonx) [reply docs/GETTING-STARTED.md]: applied

File: docs/GETTING-STARTED.md [hunk 11]
```diff
@@ -50,17 +31,21 @@ or using script which opens defualt proxy (the same as above one)
 ```bash
 ssh -D localhost:12345 <user>@<host>
 ```
-then save the address of the socket you've opened in CERN_SOCKS env var e.g. 
+then save the address of the socket you've opened in `CERN_SOCKS` env var e.g.
```
Comment (Ehevi) [inline docs/GETTING-STARTED.md:34]: The address of the socket you've opened should be saved in `CERN_SOCKS` env var, e.g.:

File: docs/GETTING-STARTED.md (hunk 9 above)
Comment (xsalonx) [reply docs/GETTING-STARTED.md]: modified

File: docs/GETTING-STARTED.md [hunk 12]
```diff
@@ -50,17 +31,21 @@ or using script which opens defualt proxy (the same as above one)
 ```bash
 ssh -D localhost:12345 <user>@<host>
 ```
-then save the address of the socket you've opened in CERN_SOCKS env var e.g. 
+The address of the socket you've opened should be saved in CERN_SOCKS env var, e.g.:
 ```bash
 export CERN_SOCKS='socks://localhost:12345'
 ```
-3. If you want to not use proxy set CERN_SOCKS to false:
+3. If you want to use proxy set `CERN_SOCKS` to true:
```
Comment (Ehevi) [inline docs/GETTING-STARTED.md:38]: *Missing comma*
If you want to use proxy, set `CERN_SOCKS` to true

File: docs/GETTING-STARTED.md [hunk 13]
```diff
@@ -50,17 +31,21 @@ or using script which opens defualt proxy (the same as above one)
...
 ```bash
-export CERN_SOCKS='false'
+export CERN_SOCKS='true'
 ```
 
-## Database
-For the sake of using the application with mock data and managing mock data changes, the `mockData.ipynb` notebook has been created. Make sure you meet the requirements:
+## Database dumps managing
```
Comment (Ehevi) [inline docs/GETTING-STARTED.md:43]: Just `Database dumps` is enough

File: docs/GETTING-STARTED.md (hunk 12 above)
Comment (xsalonx) [reply docs/GETTING-STARTED.md:38]: applied

File: docs/GETTING-STARTED.md (hunk 13 above)
Comment (xsalonx) [reply docs/GETTING-STARTED.md:43]: applied

**Review by Ehevi — APPROVED** [review]

**Pull request merged**